pkg/server/server.go

```diff
@@ -83,40 +84,57 @@ func (s *srv) CreateUser(context.Context, *proto.CreateUserRequest) (*proto.Crea
 }
 
 func (s *srv) SetPassword(ctx context.Context, in *proto.SetPasswordRequest) (*proto.SetPasswordResponse, error) {
-	pw, err := s.serverAuthPassword()
-	if err != nil {
-		return nil, fmt.Errorf("getting manage auth password: %w", err)
-	}
-	if err := connection.CheckAuthFromContext(ctx, pw); err != nil {
-		return nil, fmt.Errorf("authorization failed: %w", err)
-	}
 	ds := datastore.New(s.config.datastoreReaderURL(), s.config.datastoreWriterURL())
 	auth := auth.New(s.config.authURL())
 	return setpassword.SetPassword(ctx, in, ds, auth)
 }
 
 func (s *srv) Tunnel(ts proto.Manage_TunnelServer) error {
-	pw, err := s.serverAuthPassword()
+	return tunnel.Tunnel(ts)
+}
+
+func serverAuthUnaryInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
```

Comment:
Maybe rename this function to `authUnaryInterceptor`. The package is called `server`. So there is no need to call the function `serverFOO`. See https://go.dev/blog/package-names#naming-package-contents

This rule is not so important on unpublished functions but I think it a a good style to use it also on unpublished function names.

There are more examples here in this file.